validation/registry.go

```diff
@@ -0,0 +1,96 @@
+package validation
+
+import (
+	"fmt"
+	"github.com/UCLALibrary/validation-service/validation/checks"
+	"github.com/UCLALibrary/validation-service/validation/config"
+	"go.uber.org/zap"
+)
+
+// Registry keeps a collection of registered validators.
+type Registry struct {
+	profiles *config.Profiles
+	logger   *zap.Logger
+}
+
+// Validators is a (sub)set of validation checks and their names.
+//
+// We use a struct instead of a map to preserve validator order.
+type Validators struct {
+	Names  []string
+	Checks []Validator
+}
+
+// The constructor creates a new instance of the Validator interface.
+type constructor func(args ...interface{}) (Validator, error)
+
+// A registry of Validator constructors.
+//
+// New validators need to be manually configured in this map to avoid reflection.
```

Comment:
What does this mean to avoid reflection here?

Reply:
Another way, instead of hard-coding constructors here, would have been to use reflection to get the name/constructors of the validators from their types (maybe all of the things in the `checks` package). Go provides this feature in the `reflect` package, but my research suggests it's not commonly used in idiomatic Go code, with reasons including: 1) it has a performance overhead, 2) it undermines Go's compile time safety (leaving it to instantiate at runtime), 3) it's not as easy to read, reason, and work with code that uses it, and 4) it's just not considered idiomatic Go because it's used very sparingly in the Go world. So, I avoided it in a first pass since we're trying to learn to be more idiomatic in our use of Go.

Reply:
I recognize the registry code could get quite large, depending on how many validators we create -- that's why I put it in its own type/file. As the project develops, we may try different, more sophisticated ways of doing this without reflection.